### PR TITLE
build: Run integration tests on release branch and tags

### PR DESCRIPTION
**Pull request by travisn**
**Description of your changes:**
The integration test actions were not running on the release branch or tags due too restrictive of a condition. 
Already merged to release-1.7 with #8374 

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/master/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [ ] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.

### CODE REVIEW

File: .github/workflows/integration-tests-on-release.yaml
Comment (travisn) [inline .github/workflows/integration-tests-on-release.yaml:17]: @subhamkrai @leseb Do we need this condition for anything? This has already been merged to release-1.7 to unblock the builds, but before merging to master wanted to confirm if i'm missing something.

Comment (subhamkrai) [reply .github/workflows/integration-tests-on-release.yaml:17]: right, we can remove it as we are already running this on
```
on:
  push:
    tags:
      - v*
    branches:
      - master
      - release-* 
  ```

**Review by subhamkrai — APPROVED** [review]

**Pull request merged**
